.travis.yml

```diff
@@ -7,19 +12,39 @@ rvm:
 git:
   submodules: false
 
+addons:
+  apt:
+    packages:
+      - valgrind
+      - check
+      - pkg-config
+      - gcc
+
 before_install:
-  - sed -i 's/git@github.com:/https:\/\/github.com\//' .gitmodules
-  - git submodule update --init --recursive
-  - git config --global user.email "travis@example.com"
-  - git config --global user.name "Travis"
+  - . ./setup-travis.sh
+install:
+  - echo $(pwd)
+  - BUNDLE_GEMFILE=Gemfile bundle install --retry=3 --jobs=3 --deployment
 
 services:
   - postgresql
 
 before_script:
   - createuser -U postgres -s tmc
-
-script:
   - bundle exec rake db:reset
   - export rvmsudo_secure_path=1
-  - rvmsudo bundle exec rake spec SPEC="spec/controllers spec/models spec/helpers spec/mailers"
+
+script:
+  - env
+  - BUNDLE_GEMFILE=$(pwd)/ext/tmc-sandbox/web/Gemfile ./ext/tmc-sandbox/web/webapp.rb start
+  - BUNDLE_GEMFILE=$(pwd)/ext/tmc-sandbox/web/Gemfile ./ext/tmc-sandbox/web/webapp.rb status
+  - sleep 10
+  - BUNDLE_GEMFILE=$(pwd)/ext/tmc-sandbox/web/Gemfile ./ext/tmc-sandbox/web/webapp.rb status
+  - curl 127.0.0.1:3001
+  - curl -v 127.0.0.1:3001/status.json
```

Comment:
Instead of the sleep, could we have e.g. a for-loop that checks `$(curl http://127.0.0.1:3001/status.json)` against an expected JSON string a limited number of times and waits for 1 sec between checks?